vendor/brave-ios/scripts/mojo/mojom_objc_generator.py

```diff
@@ -128,13 +143,91 @@ def _ObjCEnumFormatter(self, str):
     return ''.join(map(lambda p: p.capitalize(), str.split('_')))
 
   def _IsObjCNumberKind(self, kind):
-    return (mojom.IsIntegralKind(kind) or mojom.IsDoubleKind(kind) or
-            mojom.IsFloatKind(kind))
+    return kind in _kind_to_nsnumber_getter
 
   def _CppPtrToObjCTransform(self, kind, obj):
     args = (self.class_prefix, kind.name, kind.name, obj)
     return "[[%s%s alloc] initWith%s:*%s]" % args
 
+  def _ObjCToCppAssign(self, field, obj):
+    kind = field.kind
+    accessor = "%s.%s" % (obj, self._ObjCPropertyFormatter(field.name))
+
+    if self._IsObjCNumberKind(kind):
+      return accessor
+    if kind in self.module.structs:
+      return "%s.cppObjPtr" % accessor
+    if mojom.IsEnumKind(kind):
+      return "static_cast<%s::%s>(%s)" % (self._CppNamespace(), kind.name, accessor)
```

Comment:
I assume this is for a regular C++98 enum and not C++11 enum-class (which so far, I haven't seen the project use).
Looks good :)

Reply:
Actually, mojo generated enum's _do_ use C++ 11 enum classes, is there something that needs to be done differently in these cases?